custom_components/animated_scenes/scene_config.py

```
@@ -0,0 +1,488 @@
+
+    Returns:
+        The trimmed list items between brackets. Malformed empty items are
+        preserved for the caller's typed validation to reject.
+    """
+    return [item.strip() for item in value.strip("][").split(",")]
+
+
+def _float_to_int(value: float) -> int:
+    """Convert a whole float to int or raise for fractional values."""
+    if not value.is_integer():
+        raise vol.Invalid("brightness must be an int")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)